From 3bb338ba72d274f2006d49816c6e9d7afbf312fc Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Date: Thu, 26 Nov 2020 10:22:46 +0200
Subject: [PATCH 11/76] drivers: clk: s32g274a: Place LLCE block under XBAR
 clock

Enable the partition block associated to LLCE IP before XBAR enablement.
This is needed because LLCE SRAM blocks are powered by XBAR_DIV2.

Issue: ALB-5829
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
---
 drivers/clk/s32/s32g274a_clk.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/clk/s32/s32g274a_clk.c b/drivers/clk/s32/s32g274a_clk.c
index e31cc29f60..8c5e4e9730 100644
--- a/drivers/clk/s32/s32g274a_clk.c
+++ b/drivers/clk/s32/s32g274a_clk.c
@@ -13,19 +13,19 @@
 #define ARR_CLK(N)	S32G274A_CLK_INDEX(N)
 
 /* XBAR_2X */
+static struct s32gen1_part_block llce_block =
+		S32GEN1_PART_BLOCK_NO_STATUS(cgm0_mux0_clk, 3,
+					     s32gen1_part_block0);
 struct s32gen1_clk xbar_2x_clk =
-		S32GEN1_CHILD_CLK(cgm0_mux0_clk, 48 * MHZ, 800 * MHZ);
+		S32GEN1_FREQ_MODULE_CLK(llce_block, 48 * MHZ, 800 * MHZ);
 
 /* PER_CLK */
 static struct s32gen1_clk per_clk =
 		S32GEN1_FREQ_MODULE_CLK(per_div, 0, 80 * MHZ);
 
 /* CAN_PE_CLK */
-static struct s32gen1_part_block llce_block =
-		S32GEN1_PART_BLOCK_NO_STATUS(cgm0_mux7_clk, 3,
-					     s32gen1_part_block0);
 static struct s32gen1_clk can_pe_clk =
-		S32GEN1_FREQ_MODULE_CLK(llce_block, 40 * MHZ, 80 * MHZ);
+		S32GEN1_CHILD_CLK(cgm0_mux7_clk, 40 * MHZ, 80 * MHZ);
 
 /* PFE_MAC0_TX_DIV */
 static struct s32gen1_fixed_clock pfe_mac0_ext_tx =
-- 
2.17.1

